vars/cloverReportPublish.groovy

```diff
@@ -0,0 +1,128 @@
+   *
+   *
+   * config['inst_rpms']           Additional rpms to install.  Optional
+   *
+   * config['NODELIST']            NODELIST of nodes to run tests on.
+   *                               Default env.NODELIST
+   *
+   * config['node_count']          Count of nodes that will actually be used
+   *                               the test.  Default will be based on the
+   *                               enviroment variables for the stage.
+   *
+   * config['stash']               Stash name for the ".build-vars.*" files.
+   */
+
+def call(Map config = [:]) {
+
+  Map stage_info = parseStageInfo(config)
+
+  String inst_rpms = config.get('inst_rpms','')
+
+  if (stage_info['target'].startsWith("centos")) {
+    inst_rpms += ' java-1.8.0-openjdk'
+  } else if (stage_info['target'].startsWith('ubuntu')) {
+    inst_rpms += ' openjdk-8-jdk'
+  } else {
+    // Assume a Suse distro
+    inst_rpms += ' java-1_8_0-openjdk'
+  }
+
+  provisionNodes NODELIST: nodelist,
+                 node_count: stage_info['node_count'],
+                 distro: stage_info['target'],
+                 inst_repos: config.get('inst_repos',''),
+                 inst_rpms: inst_rpms
+
+  String coverage_website = config.get('coverage_website',
+                                       'coverage_website.zip')
+
+  String url_base = env.JENKINS_URL +
+                    'job/daos-stack/job/tools/job/master' +
+                    '/lastSuccessfulBuild/artifact/'
+
+  httpRequest url:  url_base + 'bullseyecoverage-linux.tar',
+              httpMode: 'GET',
+              outputFile: 'bullseye.tar'
+
+  httpRequest url: url_base + 'bullshtml.jar',
+              httpMode: 'GET',
+              outputFile: 'bullshtml.jar'
+
+  def stashes = []
+  if (config['coverage_stashes']) {
+    stashes = config['coverage_stashes']
+  } else {
+    error "No coverage_stashes passed!"
+  }
+
+  String target_stash = "${stage_info['target']}-${stage_info['compiler']}"
+  if (stage_info['build_type']) {
+    target_stash += '-' + stage_info['build_type']
+  }
+  
+  unstash config.get('stash', "${target_stash}-build-vars")
+
+  int stash_cnt=0
+  stashes.each {
+    unstash it
+    stash_cnt++
+    String new_name = "test.cov_${stash_cnt}"
+    // Need plugin Upgrade
+    //fileOperations([fileRenameOperation(source: 'test.cov',
+    //                                    destination: new_name)])
+    sh label: 'Rename file',
+       script: "mv test.cov ${new_name}"
+  }
+
+  sh label: 'Create Coverage Report',
+     script: config.get('coverage_script', 'ci/bullseye_generate_report.sh')
+
+  def cb_result = currentBuild.result
+  step([$class: 'CloverPublisher',
+        cloverReportDir: 'test_coverage',
+        cloverReportFileName: 'clover.xml',
+        healthyTarget: config.get('coverage_healthy',
+                                   [methodCoverage: 70,
+                                    conditionalCoverage: 80,
```

Comment:
We're nowhere near these thresholds are we?  What happens to runs that are below the thresholds?

Reply:
Those are just the default thresholds for the CloverPublisher plugin.
Currently we are overriding them in the Jenkinsfile.
Eventually who ever is maintaining the coverage tests will need to set some reasonable defaults here, so that the Jenkinsfile overrides are not needed.

I am not sure what happens if we go below the thresh hold.  I am just setting up the framework that can be used in future PRs.